Framework/PythonInterface/plugins/algorithms/IntegratePeaksSkew.py

```diff
@@ -1069,6 +1071,14 @@ def PyExec(self):
                 optimise_xwindow,
                 threshold_i_over_sig,
             )
+
+            # get all detector IDs related to the spectrum of the peak
+            ispec = ws.getIndicesFromDetectorIDs([detid])
+            detectors_per_spec = ws.getSpectrum(ispec[0]).getDetectorIDs()
```

Comment:
For the skew case - again the shape need to be written after integration. You should be able to find what you need on the PeakData class, for example the detector IDs that contribute to the peak are `peak_data.detids[peak_data.poeak_mask]`. The TOF/d-spacing limits are given by `ws.readX(ispec)[peak_data.ixhi_opt]  - ws.readX(ispec)[peak_data.ixlo_opt]` - for weak peaks the x-lmits arent neceessarily optmised so you might have to use `ixlo` and `ixhi`.